### DISCUSSION TITLE
/articles/sharepoint-add-in-model-retirement-other-services-unpacked/

### DISCUSSION THREAD

**Discussion by giscus** [Voitanos/voitanos.io-comments#224] [Comments]
This is the definitive guide to the Microsoft 365 retirements announced in 2023: SharePoint Add-in model, SharePoint 2013 workflows, and Azure ACS.
https://www.andrewconnell.com/articles/sharepoint-add-in-model-retirement-other-services-unpacked/

**Comment by OverkillTV** [comment]
Do we have any way to retrieve the list items inside the SharePoint Hosted Add-in from outside of its isolated app ? Since we probably have to backup the data inside the SharePoint Hosted Add-in at some point during the migration process right ?

**Reply by andrewconnell** [reply]
Even though it's isolated, you can still get the data within the AppWeb as it's just a SharePoint site. So knowing the address, you should be able to use SharePoint's REST API to retrieve the data.

**Comment by euankennedy** [comment]
This is a great resource, many thanks for it.
I have two questions. First, can you confirm that there is no modern equivalent to the way provider-hosted RERs can run under the clientcontext of the user whose action triggered the event?
And second, I have recently found that it seems no longer to be possible to register an ACS principal (subject to new restrictions on appregnew.aspx) using an existing client id & secret, e.g. in the scenario I have multiple tenancies who are using a single instance of my provider-hosted RER service. That is, I can create a new ACS principal, but I can’t register one by providing details of an existing one, meaning I can no longer service new customers with my existing service. Is there something I’m missing or is it likely this is part of a gradual deprecation of ACS?

**Reply by andrewconnell** [reply]
@euankennedy said:
> First, can you confirm that there is no modern equivalent to the way provider-hosted RERs can run under the clientcontext of the user whose action triggered the event?

Correct, there's no concept of Remote Event Receivers. But if you're looking for a trigger to fire an event when an item in a list/library is created/deleted/updated, you can use a Microsoft Graph webhook... learn more 👉 **[Microsoft Graph Webhooks - What, Why, How & Best Practices](https://www.voitanos.io/blog/microsoft-graph-webhook-delta-query/)**.

@euankennedy said:
> And second, I have recently found that it seems no longer to be possible to register an ACS principal (subject to new restrictions on appregnew.aspx) using an existing client id & secret, e.g. in the scenario I have multiple tenancies who are using a single instance of my provider-hosted RER service. That is, I can create a new ACS principal, but I can’t register one by providing details of an existing one, meaning I can no longer service new customers with my existing service. Is there something I’m missing or is it likely this is part of a gradual deprecation of ACS?

That I'm not sure about (the part about registering existing credentials... but I would suggest moving away from ACS as we're just 2mo away from the first phase of its retirement. Instead, look to creating an app with Entra ID & granting the app the necessary permissions.

**Comment by sschaff647** [comment]
Any thoughts on the recent announcement for retirement of field customizers?  Is there an option any more to place custom business logic on lists or libraries now? https://support.microsoft.com/en-gb/office/support-update-for-sharepoint-framework-field-customizers-in-lists-and-document-libraries-0eccc64e-4512-47df-9da0-d855be22fb0a#:~:text=Field%20Customizers%20will%20be%20retired,function%20and%20run%20as%20expected

**Reply by andrewconnell** [reply]
Yup... I have **A LOT** ot say on this. 

I'm working on an article now that will go out later today after my monthly office hours (where I'll also address it)... if you can [join me for the livestream in about 30 minutes](https://www.voitanos.io/webinars/microsoft-365-full-stack-office-hours-2025-06-june/), cool... otherwise it will be recorded and available at the same link.

It's also the topic of my next episode on the **[Code. Deploy. Go Live.](https://codedeploygo.live/)** podcast, episode 8, out tomorrow (June 18).

**Reply by andrewconnell** [reply]
I shared my thoughts on the field customizer retirement in my latest article: **[Microsoft retires SharePoint Framework field customizers](https://www.voitanos.io/blog/sharepoint-framework-field-customizer-retirement/)**
